setup.sh

```
@@ -1,39 +1,197 @@
+# Interactive API key setup
+if [ "$API_KEY_CONFIGURED" = false ]; then
+    echo ""
+    echo -e "${YELLOW}🔑 API Key Configuration${NC}"
+    echo "To use the Google Gemini API, you'll need an API key."
+    echo "Get your key from: https://makersuite.google.com/app/apikey"
+    echo ""
+    echo -e "${BLUE}Would you like to:${NC}"
+    echo "  1) Enter your API key now"
+    echo "  2) Use mock mode (no API key required)"
+    echo ""
+    read -p "Choose option (1 or 2): " choice
```

Shellcheck (0.10.0)

[info] read without -r will mangle backslashes.

(SC2162)